backend/users/views.py

```diff
@@ -1,12 +1,27 @@
-from rest_framework.viewsets import ModelViewSet
+from rest_framework import status
+from rest_framework.mixins import RetrieveModelMixin, UpdateModelMixin
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
 
 from users.models import Profile
 from users.permissions import IsAdminOrOwner
 from users.serializers import ProfileSerializer
 
 
-class ProfileViewSet(ModelViewSet):
+class ProfileViewSet(RetrieveModelMixin, UpdateModelMixin, GenericViewSet):
     queryset = Profile.objects.all()
     serializer_class = ProfileSerializer
-    http_method_names = ["put", "get", "patch"]
     permission_classes = [IsAdminOrOwner]
+
+    def update(self, request, *args, **kwargs):
+        instance = self.get_object()
+        serializer = ProfileSerializer(
+            instance, data=request.data, partial=kwargs.pop("partial", False)
+        )
+        serializer.is_valid()
```

Comment:
Call `is_valid(raise_exception=True)`, without it, passing invalid data won't raise exception